src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/simple/DPBasedJoinPlanOptimizer.java

```diff
@@ -82,8 +87,50 @@ else if ( plan_left.getRootOperator() instanceof PhysicalOpRequestWithTranslatio
 
     }
 
-    // This method returns all subsets (with the given size) of the given superset.
-    public static <T> List<List<T>> getSubSet( final List<T> superset, final int n ){
+    public abstract <T> List<Pair<List<T>, List<T>>> splitIntoSubSets( final List<T> superset );
+
+    /**
+     * In cases in which there is a union with requests under right input,
+     * this function turns the requests into xxAdd operators with the previous join arguments as subplans.
+     **/
+    protected PhysicalPlan createPlansWithUnaryOpForUnionPlan( final PhysicalPlan inputPlan, final PhysicalPlan unionPlan ) {
```

Comment:
Why is this function called "createPlan**s**..." when it returns only a single plan?